pkg/app/api/grpcapi/web_api.go

```diff
@@ -532,6 +534,23 @@ func (a *WebAPI) ListPipeds(ctx context.Context, req *webservice.ListPipedsReque
 		return nil, status.Error(codes.Internal, "Failed to get pipeds")
 	}
 
+	// Check piped connection status if necessary.
+	// The connection status of piped determined by its submitted stat in pipedStatCache.
+	if req.WithStatus {
+		for i := range pipeds {
+			_, err := a.pipedStatCache.Get(pipeds[i].Id)
+			if errors.Is(err, cache.ErrNotFound) {
+				pipeds[i].Status = model.Piped_OFFLINE
+				continue
+			}
+			if err != nil {
+				pipeds[i].Status = model.Piped_UNKNOWN
+				continue
+			}
+			pipeds[i].Status = model.Piped_ONLINE
+		}
+	}
+
```

Comment:
Btw, the current web implementation is not showing that status yet, right?
It would be nice to have it.

Reply:
Yes, I will address it with a separated PR (the UI)

Reply:
👍